test/svg-sprite/mode/base.test.js

```diff
@@ -0,0 +1,438 @@
+            base._buildHTMLExample(TEST_FILES, testFn);
+
+            expect(testFn).toHaveBeenCalledWith(null, base.data);
+            expect(TEST_FILES).toStrictEqual({
+                example: new File({
+                    base: TEST_SPRITER.config.dest,
+                    path: TEST_CONFIG.example.dest,
+                    contents: Buffer.from('test example')
+                })
+            });
+            expect(TEST_SPRITER.verbose).toHaveBeenCalledWith('Created «%s» HTML example file', TEST_MODE_NAME);
+        });
+    });
+
+    describe('testing _addUnit()', () => {
+        it('should add unit to number if number is not zero', () => {
+            expect.hasAssertions();
+            expect(SVGSpriteBase.prototype._addUnit(1, 'px')).toBe('1px');
+        });
+
+        it('should not add anything if number is zero', () => {
+            expect.hasAssertions();
+            expect(SVGSpriteBase.prototype._addUnit(0, 'px')).toBe('0');
+        });
+    });
+
+    describe('testing declaration()', () => {
+        it('should return local declaration if exists', () => {
+            expect.hasAssertions();
+
+            const TEST_DECLARATION = 'test declaration';
+
+            expect(SVGSpriteBase.prototype.declaration(true, TEST_DECLARATION)).toBe(TEST_DECLARATION);
+        });
+
+        it('should return empty string if local declaration is falsy', () => {
+            expect.hasAssertions();
+            expect(SVGSpriteBase.prototype.declaration(true, false)).toBe('');
+        });
+
+        it('should return global if provided', () => {
+            expect.hasAssertions();
+
+            const TEST_DECLARATION = 'test declaration';
+
+            expect(SVGSpriteBase.prototype.declaration(`${TEST_DECLARATION} `)).toBe(TEST_DECLARATION);
+        });
+
+        it('should return empty string if global declaration is not provided', () => {
+            expect.hasAssertions();
+            expect(SVGSpriteBase.prototype.declaration()).toBe('');
+        });
+    });
+
+    describe('testing _addCacheBusting()', () => {
+        it('should add busting', () => {
+            expect.hasAssertions();
+
+            const TEST_SPRITER = {
+                config: {
+                    dest: '.',
+                    shape: {
+                        spacing: []
+                    }
+                },
+                _limit: 100,
+                debug: jest.fn(),
+                verbose: jest.fn()
+            };
+            const TEST_CONFIG = {
+                dest: '.',
+                bust: true,
+                prefix: '1',
+                sprite: '',
+                example: {
+                    dest: '.'
+                }
+            };
+            const { cls } = getClassAndInitFn();
+            const base = new cls(TEST_SPRITER, TEST_CONFIG, {}, '');
+            const TEST_SVG = 'svg';
+            const TEST_SPRITE = 'sprite-ae8eb96d.svg';
```

Comment:
I'd just test for the number of characters here, like `^sprite-\w{8}\.svg$`. I prefer not using regex when possible but I think it makes sense here.

Reply:
BTW, unsure why we use 8 characters. I was under the impression 7 was more common.